### PR TITLE
Deployment and Finetuning config

### PR DESCRIPTION
**Pull request by VipulMascarenhas**
This PR covers the following:

1. MODEL_BY_REFERENCE_OSS_PATH_KEY has been changed from `Object Storage Path` to `artifact_location`.
2. load_config moved to utils so that it can be reused for other config files as well.
3. Finetuning config has been added.  To call, use `GET http://localhost:8888/aqua/finetuning/config/<OCID>`

`finetuning_config.json`
```
{
"shape": {
      "VM.GPU.A10.1": {
        "replica": 1,
        "batch_size": 2
    }
}
```

5. Deployment config will now have nested parameters required for deployment, instead of common one at the root level. The config will look like this:

`deployment_config.json`
```
{
    "shape": [
        "VM.GPU.A10.1",
        "VM.GPU.A10.2"
    ]
}
```

### CODE REVIEW

File: ads/aqua/extension/finetune_handler.py [hunk 1]
```diff
@@ -39,5 +55,13 @@ def post(self, *args, **kwargs):
         except Exception as ex:
             raise HTTPError(500, str(ex))
```
Comment (lu-ohai) [inline ads/aqua/extension/finetune_handler.py:56]: This is a bug here. `create` inside will raise 400 error, but adding this `try catch` will make all client errors from `create` as 500. This needs to be removed as the `handle_exceptions` will take care of this. Same for the deployment handler to create model deployment.

File: ads/aqua/deployment.py [hunk 2]
```diff
@@ -252,10 +248,27 @@ def create(
...
+        env_var.update({"MODEL": f"{AQUA_MODEL_DEPLOYMENT_FOLDER}{model_path_prefix}"})
+
+        deployment_config = load_config(
+            artifact_path,
+            config_file_name=AQUA_MODEL_DEPLOYMENT_CONFIG,
+            auth=self._auth,
+        )
+        shape_dict = deployment_config["shape"]
+        if instance_shape not in shape_dict:
+            raise AquaValueError(
```
Comment (lu-ohai) [inline ads/aqua/deployment.py:266]: Same as I mentioned below, if 400 client error is raised in create model deployment, the `try catch` must be removed.

File: ads/aqua/extension/finetune_handler.py (hunk 1 above)
Comment (VipulMascarenhas) [reply ads/aqua/extension/finetune_handler.py:56]: yes, you're right. post method hasn't been implemented yet, we can address this change when in the create FT PR.

Comment (lu-ohai) [reply ads/aqua/extension/finetune_handler.py:56]: Same for the `deployment_handler.py` as there will be 400 error raised in `create`: https://github.com/oracle/accelerated-data-science/blob/shape_config_api/ads/aqua/extension/deployment_handler.py#L111

Comment (VipulMascarenhas) [reply ads/aqua/extension/finetune_handler.py:56]: updated this PR for all apis within deployment_handler, can you check if anything is missing?

File: ads/aqua/deployment.py (hunk 2 above)
Comment (VipulMascarenhas) [reply ads/aqua/deployment.py]: updated

**Pull request merged**